backend/codingchallenge/api/views/admin/adminApplicationView.py

```diff
@@ -0,0 +1,104 @@
+import math
+import time
+import random
+
+# Authentication imports
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+from django.contrib.auth.models import User
+
+# RESTapi imports
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+# import model
+from ...models import Application, Challenge
+
+# import serializer
+from ...serializers import ApplicationSerializer
+
+
+#
+
+### endpoint: /api/admin/applications
+class AdminApplicationView(APIView):
+    # grant permission only for admin user
+    permission_classes = [IsAdminUser]
+
+    name = "Admin Application View"
+    description = "handling all requests for applications as a admin"
+
+    # 1. List all
+    def get(self, request, *args, **kwargs):
+        '''
+        get Application with follwing columns:
+            applicationId,
+            operatingSystem,
+            programingLanguage,
+            expiry,
+            submission,
+            githubRepo,
+            status,
+            created,
+            modified
+        '''
+        applications = Application.objects
+        serializer = ApplicationSerializer(applications, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    # 2. Create
+    def post(self, request, *args, **kwargs):
+        '''
+        post Application with
+            required arguments:
+                applicationId,
+                applicantEmail
+
+            optional arguments:
+                challengeId
+                days
+        '''
+
+        try:
+            # random challenge
+            challengeId = random.choice(Challenge.objects.all()).id
+            # default 2 days time for start
+            days = 2
+
+        except IndexError:
+            return Response({'detail': 'there are no challenges in database'},
+                            status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        if len(request.data) == 0:
+            return Response({'detail': 'body is empty'}, status=status.HTTP_204_NO_CONTENT)
+
+        try:
+
+            if Application.objects.all().filter(
+                    applicationId=request.data.get('application').get('applicationId')):
```

Comment:
API-Doku updated. Now not specific objects like 'application' shall be returned but instead objects without name. Applies for all following requests, that look like this: request.data.get('application')

Reply:
will be implemented in #85